### PR TITLE
feat: stand alone running event hash

### LINKED ISSUE #12656: Create a wiring component for computing the running event hash
In order to run turtle tests, we need to have all things running in the wiring framework.

We are deleting the consensus event stream soon-ish, so don't want to spend effort rewriting the CES so that it's compatible with the wiring framework.

Most of what the CES does is not super important, but it currently fills the critical role of computing the running event hash. Even when the CES is gone, we still want to compute a running event hash.

Create a platform component whose job it is to compute the running event hash. The running event hash computed by the CES can be ignored. Then, when we do turtle tests, we can simply turn off the CES.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12656

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java [hunk 1]
```diff
@@ -381,7 +381,8 @@ private static ReservedSignedState handleNextRound(
         final PlatformState platformState = newState.getPlatformState();
 
         platformState.setRound(round.getRoundNum());
-        platformState.setRunningEventHash(getHashEventsCons(previousState.get().getHashEventsCons(), round));
+        platformState.setLegacyRunningEventHash(getHashEventsCons(
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java:384]: Do you also need to set the new running hash here?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ private static PlatformState buildGenesisPlatformState(
 
         platformState.setCreationSoftwareVersion(appVersion);
         platformState.setRound(0);
-        platformState.setRunningEventHash(null);
+        platformState.setLegacyRunningEventHash(null);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java:50]: do you also need to set the new running hash here?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/PlatformStateUtils.java [hunk 3]
```diff
@@ -52,7 +52,7 @@ public static PlatformState randomPlatformState(final Random random) {
                 .build();
 
         platformState.setAddressBook(addressBook);
-        platformState.setRunningEventHash(randomHash(random));
+        platformState.setLegacyRunningEventHash(randomHash(random));
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/PlatformStateUtils.java:55]: Do you also need to set the new running hash here?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java [hunk 4]
```diff
@@ -152,6 +163,7 @@ private PlatformState(final PlatformState that) {
         this.addressBook = that.addressBook == null ? null : that.addressBook.copy();
         this.previousAddressBook = that.previousAddressBook == null ? null : that.previousAddressBook.copy();
         this.round = that.round;
+        this.legacyRunningEventHash = that.legacyRunningEventHash;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java:166]: copy new running hash here

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java [hunk 5]
```diff
@@ -130,6 +130,8 @@ public record PlatformSchedulersConfig(
         @ConfigProperty(defaultValue = "500") int shadowgraphUnhandledCapacity,
         @ConfigProperty(defaultValue = "SEQUENTIAL_THREAD") TaskSchedulerType consensusRoundHandlerSchedulerType,
         @ConfigProperty(defaultValue = "5") int consensusRoundHandlerUnhandledCapacity,
+        @ConfigProperty(defaultValue = "SEQUENTIAL CAPACITY(5) UNHANDLED_TASK_METRIC BUSY_FRACTION_METRIC")
+                TaskSchedulerConfiguration runningEventHasher,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:134]: missing doc

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ConsensusRoundHandlerWiring.java [hunk 6]
```diff
@@ -32,7 +32,7 @@
  * Wiring for the {@link com.swirlds.platform.eventhandling.ConsensusRoundHandler}
  *
  * @param roundInput              the input wire for consensus rounds to be applied to the state
- * @param runningHashUpdateInput  the input wire for updating the running event hash
+ * @param overrideLegacyRunningEventHashInput  the input wire for updating the running event hash
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ConsensusRoundHandlerWiring.java:35]: doc alignment

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/RandomSignedStateGenerator.java [hunk 7]
```diff
@@ -184,7 +184,7 @@ public SignedState build() {
         final PlatformState platformState = stateInstance.getPlatformState();
 
         platformState.setRound(roundInstance);
-        platformState.setRunningEventHash(hashEventsConsInstance);
+        platformState.setLegacyRunningEventHash(hashEventsConsInstance);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/RandomSignedStateGenerator.java:187]: Should new running hash be set here?

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java [hunk 8]
```diff
@@ -129,7 +129,7 @@ void normalOperation() throws InterruptedException {
         verify(swirldStateManager).handleConsensusRound(consensusRound);
         verify(swirldStateManager, never()).savedStateInFreezePeriod();
         verify(platformState)
-                .setRunningEventHash(
+                .setLegacyRunningEventHash(
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java:132]: The `ConsensusRoundHandlerTests` should probably be expanded to cover the new running hash functionality

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java:384]: I updated this to just set the null hash (i.e. `Cryptography.getNullHash())`). This recovery workflow already does not produce a state with the exact same hash as the original, so I feel no stress by making the result different in more ways.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java:50]: Good idea, set this equal to the null hash. I was handling the situation where this field was null for a genesis state, but this is more elegant. Once we migrate mainnet, I can completely remove the logic that currently handles an initial state with a `null` value here.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java:166]: I already do, see line 167

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:134]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ConsensusRoundHandlerWiring.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ConsensusRoundHandlerWiring.java:35]: fixed

**Review by alittley — APPROVED** [review]

**Pull request merged**